src/python/pants/engine/target.py

```diff
@@ -997,12 +997,19 @@ class TransitiveTargetsRequest:
 
     roots: Tuple[Address, ...]
     include_special_cased_deps: bool
+    # If this predicate returns false, then the target's dependencies will be ignored.
+    do_traverse_deps_predicate: Callable[[Target], bool] | None
 
     def __init__(
-        self, roots: Iterable[Address], *, include_special_cased_deps: bool = False
+        self,
+        roots: Iterable[Address],
+        *,
+        include_special_cased_deps: bool = False,
+        do_traverse_deps_predicate: Callable[[Target], bool] | None = None,
```

Comment:
:+1: on `should_` instead of `do_`. I'll change that.

This is a performance sensitive bit of code that @stuhood has talked about moving to rust. So, I used the `None` check to minimize any performance impacts (eg adding extra function calls) unless necessary.

Reply:
I renamed `do_` -> `should_`.

I left the `None` check, though, so that only requests that explicitly include a predicate incur the cost of running it.

Reply:
OK. This no longer applies to the latest version of the PR.